From 70411e53807c84fd363066f6a0f04b3790fdc6ad Mon Sep 17 00:00:00 2001
From: Aaron Williams <aaron.williams@cavium.com>
Date: Wed, 8 May 2019 20:00:16 +0200
Subject: [PATCH 300/345] Octeon: MTD: NAND: Do not call
 is_vmalloc_or_module_addr()

This function is not available to modules and is not required to be called.

Bug# 32643

Signed-off-by: Aaron Williams <aaron.williams@cavium.com>
---
 arch/mips/cavium-octeon/octeon-nand.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/arch/mips/cavium-octeon/octeon-nand.c b/arch/mips/cavium-octeon/octeon-nand.c
index 12f01e2cf1bd..57285cba08f9 100644
--- a/arch/mips/cavium-octeon/octeon-nand.c
+++ b/arch/mips/cavium-octeon/octeon-nand.c
@@ -622,10 +622,6 @@ static int octeon_nand_bch_calculate_ecc_internal(struct octeon_nand *priv,
 	int i;
 	static uint8_t *ecc_buffer;
 
-	/* Can only use logical or xkphys pointers */
-	WARN_ON(is_vmalloc_or_module_addr(buf));
-	WARN_ON(is_vmalloc_or_module_addr(code));
-
 	if (!ecc_buffer)
 		ecc_buffer = kmalloc(1024, GFP_KERNEL);
 	if ((ulong)buf % 8)
@@ -756,9 +752,6 @@ static int octeon_nand_bch_correct(struct mtd_info *mtd, u_char *dat,
 	static int buffer_size;
 	int max_time = 100;
 
-	/* Can only use logical or xkphys pointers */
-	WARN_ON(is_vmalloc_or_module_addr(dat));
-
 	if (i > buffer_size) {
 		kfree(data_buffer);
 		data_buffer = kmalloc(i, GFP_KERNEL);
-- 
2.25.1

